ext/gmp/gmp.c

```diff
@@ -381,6 +384,10 @@ static int gmp_do_operation_ex(zend_uchar opcode, zval *result, zval *op1, zval
 		DO_BINARY_UI_OP_EX(mpz_tdiv_q, gmp_mpz_tdiv_q_ui, 1);
 	case ZEND_MOD:
 		DO_BINARY_UI_OP_EX(mpz_mod, gmp_mpz_mod_ui, 1);
+		if (UNEXPECTED(EG(exception))) {
+			return FAILURE;
```

Comment:
Is it actually necessary to return failure on exception here? I think if we throw, this should still be success, otherwise we're going to try calling do_operation on the second operand again.

Reply:
I forgot to drop this again but the ``DO_BINARY_UI_OP_EX`` macro does exactly that:
```c
#define DO_BINARY_UI_OP_EX(op, uop, check_b_zero)       \
	gmp_zval_binary_ui_op(                              \
		result, op1, op2, op, uop, check_b_zero         \
	);                                                  \
	if (UNEXPECTED(EG(exception))) { return FAILURE; }  \
	return SUCCESS;
```